### PR TITLE
[virtual-wan] `az network vhub`: Add new subgroup `route-map` to support managing virtual hub route map

### PR DESCRIPTION
**Pull request by PARADISSEEKR**
---

This checklist is used to make sure that common guidelines for a pull request are followed.
close https://github.com/Azure/azure-cli/issues/24061

### Related command

`az network vhub route-map`

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
virtual-wan

File: src/virtual-wan/HISTORY.rst [hunk 1]
```diff
@@ -3,6 +3,17 @@
 Release History
 ===============
 
+0.2.16
+++++++
+* `az network vhub`: Add new subgroup `route-map` to support managing virtual hub route map.
+* Bump api version from 2021-08-01 to 2022-07-01.
```
Comment (jsntcy) [inline src/virtual-wan/HISTORY.rst:9]:
```suggestion
```

Comment (PARADISSEEKR) [reply src/virtual-wan/HISTORY.rst:9]: `az network vhub connection`, `az network p2s-vpn-gateway` and `network vpn-gateway` have not migrated to codegen v2, so the new parameters(`associated_inbound_routemap` and `associated_outbound_routemap`) use the new sdk.

Comment (necusjz) [reply src/virtual-wan/HISTORY.rst:9]: IMO, @jsntcy's point is API version should be transparent to the users.

**Review by necusjz — APPROVED** [review]

**Pull request merged**